src/diffusers/pipelines/ip_adapter/attention_processor.py

```diff
@@ -0,0 +1,406 @@
+import warnings
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class IPAttnProcessor(nn.Module):
+    r"""
+    Attention processor for IP-Adapater.
+    Args:
+        hidden_size (`int`):
+            The hidden size of the attention layer.
+        cross_attention_dim (`int`):
+            The number of channels in the `encoder_hidden_states`.
+        text_context_len (`int`, defaults to 77):
+            The context length of the text features.
+        scale (`float`, defaults to 1.0):
+            the weight scale of image prompt.
+    """
+
+    def __init__(self, hidden_size, cross_attention_dim=None, text_context_len=77, scale=1.0):
+        super().__init__()
+
+        self.hidden_size = hidden_size
+        self.cross_attention_dim = cross_attention_dim
+        self.text_context_len = text_context_len
+        self.scale = scale
+
+        self.to_k_ip = nn.Linear(cross_attention_dim or hidden_size, hidden_size, bias=False)
+        self.to_v_ip = nn.Linear(cross_attention_dim or hidden_size, hidden_size, bias=False)
+
+    def __call__(
+        self,
+        attn,
+        hidden_states,
+        encoder_hidden_states=None,
+        attention_mask=None,
+        temb=None,
+        scale=1.0,
+    ):
+        if scale != 1.0:
+            warnings.warn("`scale` of IPAttnProcessor should be set by " "`IPAdapterPipeline.set_scale`")
```

Comment:
Can't we use the `cross_attention_kwargs` mechanism here?

Reply:
Currently, IPAttnProcessor only exists in cross attention layers. `cross_attention_kwargs` will be pass into ALL AttnProcessor, 
passing extra parameters to the attention processor in self-attention will result in an error.

Maybe the standard processors could be modified so that they accept redundant parameters?

for example, change

https://github.com/huggingface/diffusers/blob/1037287e2bae30e33f844786f5e65ca5a69c41a1/src/diffusers/models/attention_processor.py#L546-L554

to 

``` python
def __call__(
        self,
        attn: Attention,
        hidden_states,
        encoder_hidden_states=None,
        attention_mask=None,
        temb=None,
        scale=1.0,
        *_args, 
        **_kwargs
    ):
```